static/app/views/ddm/useGetChartPalette.tsx

```diff
@@ -0,0 +1,62 @@
+import {useCallback, useRef} from 'react';
+
+import theme from 'sentry/utils/theme';
+
+const CACHE_SIZE = 20; // number of palettes to cache
+
+// We do length - 2 to be aligned with the colors in other parts of the app (copy-pasta)
+// We use Math.max to avoid numbers < -1 as then `getColorPalette` returns undefined (not typesafe because of array access)
+export function getChartColors(seriesCount: number) {
+  return theme.charts.getColorPalette(Math.max(seriesCount - 2, -1));
+}
+
+export function getCachedChartPalette(
+  cache: Record<string, string>[],
+  seriesNames: string[]
+): (seriesName: string) => string {
+  // We do length - 2 to be aligned with the colors in other parts of the app (copy-pasta)
+  // We use Math.max to avoid numbers < -1 as then `getColorPalette` returns undefined (not typesafe because of array access)
+  const defaultColors = getChartColors(seriesNames.length);
+
+  // Check if we already have a palette that includes all of the given seriesNames
+  // We reverse the cache to get the most recent palettes first
+  const cachedPalette = [...cache]
+    .reverse()
+    .find(palette => seriesNames.every(seriesName => seriesName in palette));
+
+  if (cachedPalette) {
+    // Shift the palette to the end of the cache, so we only remove the oldest palette when hitting the cache limit
+    cache.shift();
```

Comment:
nit(bc it looks like the cache is small): shift is an On operation. A list ds would have been preferable, but not an issue since it looks like the cache is going to be small anyways

Reply:
If you only ever use it for 20 colors, it will never be an issue. What sometimes happens is that code like this might get reused or copied for a different use-case. I wouldn't worry about it, I just wanted to highlight it. Might be worth leaving a comment there